src/LinkedListTest.java

```diff
@@ -26,21 +54,38 @@ void testAddNotFirstElement() throws Exception {
         assertEquals(node1.getNextElement(), node2);
     }
 
+    @Test
+    void testReverse() throws Exception{
+        LinkedList list = new LinkedList();
+        list.addElement(new Node(0));
+        list.addElement(new Node(1));
+        list.addElement(new Node(2));
+        list.addElement(new Node(3));
+        LinkedList reverse = new LinkedList();
+        reverse=list.reverse();
+        if (list.getLengthOfList()==0){
```

Comment:
as I can see, the length of the list in constant and equals 4 , so I have no idea why do you have so many if/else statements